### PR TITLE
codec: Use official encoding/json instead of json-iterator. (#157)

### LINKED ISSUE #157: proposal: codec:  remove json-iterator and use official encoding/json
The performance of official `encoding/json`. The saying that "official json is slow" is out-of-day. Besides, json-iterator use some low-level packages those may panic when the major version of go advances.

Json-iterator panic issues:

- [it will panic fault when Marshal map (go version 1.18) ](https://github.com/json-iterator/go/issues/608)
- [升级go1.18版本json-iterator coredump问题](https://blog.csdn.net/peng314899581/article/details/125939850)

Json-iterator performance issues:

- [JSON-ITERATOR 使用要注意的大坑](https://vearne.cc/archives/433): "在使用json-iterator的过程中，没有发现有json-iterator有内存泄露问题，可以放心使用，但在Go1.10以后性能与标准库差异不大，这个第三方库的意义已经不大了"
- [Go 语言原生的 json 包有什么问题？如何更好地处理 JSON 数据？](https://cloud.tencent.com/developer/article/1820473): "jsoniter 并没有什么明显优势。即便是 jsoniter 作为卖点的大数据量解析，优势也是微乎其微"
- [Go总结（二一）| JSON编解码](https://chende.ren/2021/06/23140319-020-runtime.html): "用json-iterator意义不大，还是用标准库吧，真要提高性能，再想想别的办法，或者研究一下上面恐怖的sonic"
- [有篇文章说“在Go1.10以后性能与标准库差异不大，这个第三方库的意义已经不大了”，是真的吗？有大佬说一下吗？](https://github.com/json-iterator/go/issues/333)

If you really have performance concern, please use sonic, NOT json-iterator. If not, choose the official one. 

Besides, less dependency, less safety issues.

### PR DESCRIPTION
**Pull request by Andrew-M-C**
Fixes #157 

RELEASE NOTES: Please refer to #157, I am also the issue author.

### CODE REVIEW

**Comment by Andrew-M-C** [comment]
I have read the CLA Document and I hereby sign the CLA

File: go.mod
Comment (liuzengh) [inline go.mod:1]: Please update the go.mod file in the `/test` directory at the same time.

**Comment by WineChord** [comment]
I modified and ran the benchmark they provided, here are the results:

```shell
[wineguo@VM-116-92-tencentos go-benchmark]$ go test -run ^$ -bench ^Bench.*Small$ . 
goos: linux
goarch: amd64
pkg: github.com/json-iterator/go-benchmark
cpu: Intel(R) Xeon(R) Platinum 8255C CPU @ 2.50GHz
BenchmarkJsonParserSmall-10              1560852               753.5 ns/op            80 B/op          4 allocs/op
BenchmarkJsnoiterPullSmall-10            2083582               584.3 ns/op            80 B/op         10 allocs/op
BenchmarkJsnoiterReflectSmall-10         1786041               669.3 ns/op           176 B/op          3 allocs/op
BenchmarkEncodingJsonStructSmall-10       402614              2826 ns/op             400 B/op          8 allocs/op
BenchmarkEasyJsonSmall-10                1602778               746.4 ns/op            64 B/op          2 allocs/op
PASS
ok      github.com/json-iterator/go-benchmark   8.763s
[wineguo@VM-116-92-tencentos go-benchmark]$ go test -run ^$ -bench ^Bench.*Medium$ . 
goos: linux
goarch: amd64
pkg: github.com/json-iterator/go-benchmark
cpu: Intel(R) Xeon(R) Platinum 8255C CPU @ 2.50GHz
BenchmarkDecodeStdStructMedium-10                  55962             21655 ns/op             512 B/op         12 allocs/op
BenchmarkEncodeStdStructMedium-10                1449410               834.3 ns/op           216 B/op          2 allocs/op
BenchmarkDecodeJsoniterStructMedium-10            207370              5563 ns/op             384 B/op         41 allocs/op
BenchmarkEncodeJsoniterStructMedium-10           1619971               734.6 ns/op           216 B/op          2 allocs/op
BenchmarkDecodeEasyJsonMedium-10                  107053             11232 ns/op             160 B/op          4 allocs/op
BenchmarkEncodeEasyJsonMedium-10                 2031669               597.1 ns/op           600 B/op          4 allocs/op
PASS
ok      github.com/json-iterator/go-benchmark   9.772s
[wineguo@VM-116-92-tencentos go-benchmark]$ go test -run ^$ -bench ^Bench.*Large$ . 
goos: linux
goarch: amd64
pkg: github.com/json-iterator/go-benchmark
cpu: Intel(R) Xeon(R) Platinum 8255C CPU @ 2.50GHz
BenchmarkJsonParserLarge-10                30417             39430 ns/op               0 B/op          0 allocs/op
BenchmarkJsoniterLarge-10                  13920             86405 ns/op           12496 B/op       1133 allocs/op
BenchmarkEncodingJsonLarge-10               4906            240958 ns/op             440 B/op         10 allocs/op
PASS
ok      github.com/json-iterator/go-benchmark   4.888s
[wineguo@VM-116-92-tencentos go-benchmark]$ 
```

The code is available at my fork: https://github.com/WineChord/go-benchmark/tree/to_mod

**Comment by WineChord** [comment]
Discussion takes place at https://github.com/trpc-group/trpc-go/issues/157.

File: go.mod
Comment (Andrew-M-C) [reply go.mod:1]: done

**Comment by Andrew-M-C** [comment]
Close due to discussion in #157

**Pull request closed without merging**